src/main/java/org/jenkinsci/plugins/github_branch_source/DefaultGitHubNotificationStrategy.java

```diff
@@ -0,0 +1,124 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.model.Job;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import jenkins.scm.api.*;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+import org.kohsuke.github.GHCommitState;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @since TODO
+ */
+public class DefaultGitHubNotificationStrategy extends AbstractGitHubNotificationStrategy {
+    public DefaultGitHubNotificationStrategy() {}
+
+    /**
+     * @since TODO
+     */
+    protected String generateContext(GitHubNotificationContext notificationContext, TaskListener listener) {
+        SCMHead head = notificationContext.getHead();
+        if (head instanceof PullRequestSCMHead) {
+            if (((PullRequestSCMHead) head).isMerge()) {
+                return "continuous-integration/jenkins/pr-merge";
+            } else {
+                return "continuous-integration/jenkins/pr-head";
+            }
+        } else {
+            return "continuous-integration/jenkins/branch";
+        }
+    }
+
+    /**
+     * @since TODO
+     */
+    protected String generateUrl(GitHubNotificationContext notificationContext, TaskListener listener) {
+        Run<?, ?> build = notificationContext.getBuild();
+        Job<?, ?> job = notificationContext.getJob();
+        String url = null;
+        try {
+            if (null != build) {
+                url = DisplayURLProvider.get().getRunURL(build);
+            }
+            else if (null != job) {
+                url = DisplayURLProvider.get().getJobURL(job);
+            }
+        } catch (IllegalStateException e) {
+            listener.getLogger().println(
+                    "Can not determine Jenkins root URL. Commit status notifications are disabled "
+                            + "until a root URL is"
+                            + " configured in Jenkins global configuration.");
+        }
+        return url;
+    }
+
+    /**
+     * @since TODO
+     */
+    protected String generateMessage(GitHubNotificationContext notificationContext, TaskListener listener) {
```

Comment:
would it make more sense to move this to a `getDefaultMessage()` method on `GitHubNotificationContext`?